dspace-spring-rest/src/test/java/org/dspace/app/rest/builder/WorkspaceItemBuilder.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.builder;
+
+import org.apache.log4j.Logger;
+import org.dspace.content.Collection;
+import org.dspace.content.WorkspaceItem;
+import org.dspace.content.service.WorkspaceItemService;
+import org.dspace.core.Context;
+
+/**
+ * Builder to construct WorkspaceItem objects
+ *
+ **/
+public class WorkspaceItemBuilder extends AbstractBuilder<WorkspaceItem, WorkspaceItemService> {
+
+    /* Log4j logger*/
+    private static final Logger log = Logger.getLogger(AbstractDSpaceObjectBuilder.class);
+
+    private WorkspaceItem workspaceItem;
+
+    protected WorkspaceItemBuilder(Context context) {
+        super(context);
+    }
+
+    public static WorkspaceItemBuilder createWorkspaceItem(final Context context, final Collection col) {
+        WorkspaceItemBuilder builder = new WorkspaceItemBuilder(context);
+        return builder.create(context, col);
+    }
+
+    private WorkspaceItemBuilder create(final Context context, final Collection col) {
+        this.context = context;
+
+        try {
+            workspaceItem = workspaceItemService.create(context, col, false);
+        } catch (Exception e) {
+            return handleException(e);
+        }
+
+        return this;
+    }
+
+    protected <B> B handleException(final Exception e) {
```

Comment:
Could you move the `org.dspace.app.rest.builder.AbstractDSpaceObjectBuilder#handleException` to the `org.dspace.app.rest.builder.AbstractBuilder` class? That why you don't have to implement this method yourself and you can reuse the existing code.